### ISSUE TITLE
`:discouraged-namespace` and `:discouraged-var` don't work if their namespaces match any `:ns-groups`

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#1942]
**version**

```
clj-kondo v2022.12.10
```

**platform**

Native version installed with `brew` on Linux (WSL 2)

**problem**

The `:discouraged-namespaces` linter does not generate warnings for any namespaces that match one of the `:ns-groups`. 

**repro**

<details>
<summary>This works as expected</summary>

```clj
{:ns-groups
 [{:pattern "^metabase\\.db\\..*"
   :name    db-namespaces}]

 :config-in-ns
 {db-namespaces
  {:linters
   {:discouraged-namespace
    {clojure.java.jdbc {:message "Use next.jdbc instead of clojure.java.jdbc"}}}}}}
```

```sh
$ clj-kondo --parallel --lint src/metabase/db
src/metabase/db/jdbc_protocols.clj:6:5: warning: Use next.jdbc instead of clojure.java.jdbc
src/metabase/db/liquibase.clj:4:5: warning: Use next.jdbc instead of clojure.java.jdbc
src/metabase/db/update_h2.clj:6:5: warning: Use next.jdbc instead of clojure.java.jdbc
linting took 110ms, errors: 0, warnings: 3
```

</details>

<details>
<summary>It stops working if you create a new group that matches the namespace you're trying to discourage</summary>

```clj
{:ns-groups
 [{:pattern "^metabase\\.db\\..*"
   :name    db-namespaces}
  {:pattern "^clojure\\.java\\.jdbc.*"
   :name    jdbc-legacy}]

 :config-in-ns
 {db-namespaces
  {:linters
   {:discouraged-namespace
    {clojure.java.jdbc {:message "Use next.jdbc instead of clojure.java.jdbc"}}}}}}
```

```sh
$ clj-kondo --parallel --lint src/metabase/db
linting took 109ms, errors: 0, warnings: 0
```

</details>

<details>
<summary>If does seem to work if you use the group in :discouraged-namespace</summary>

```clj
{:ns-groups
 [{:pattern "^metabase\\.db\\..*"
   :name    db-namespaces}
  {:pattern "^clojure\\.java\\.jdbc.*"
   :name    jdbc-legacy}]

 :config-in-ns
 {db-namespaces
  {:linters
   {:discouraged-namespace
    {jdbc-legacy {:message "Use next.jdbc instead of clojure.java.jdbc"}}}}}}
```

```sh
$ clj-kondo --parallel --lint src/metabase/db
src/metabase/db/jdbc_protocols.clj:6:5: warning: Use next.jdbc instead of clojure.java.jdbc
src/metabase/db/liquibase.clj:4:5: warning: Use next.jdbc instead of clojure.java.jdbc
src/metabase/db/update_h2.clj:6:5: warning: Use next.jdbc instead of clojure.java.jdbc
linting took 110ms, errors: 0, warnings: 3
```

</details>

<details>
<summary>But if multiple groups match then it stops giving warnings</summary>

```clj
{:ns-groups
 [{:pattern "^metabase\\.db\\..*"
   :name    db-namespaces}
  {:pattern "^clojure\\."
   :name    clojure-core-namespaces}
  {:pattern "^clojure\\.java\\.jdbc.*"
   :name    jdbc-legacy}]

 :config-in-ns
 {db-namespaces
  {:linters
   {:discouraged-namespace
    {jdbc-legacy {:message "Use next.jdbc instead of clojure.java.jdbc"}}}}}}

```

```sh
$ clj-kondo --parallel --lint src/metabase/db
linting took 109ms, errors: 0, warnings: 0
```

</details>

**config**

It is reproducible with just the minimal config posted above.

**expected behavior**

Namespaces in `:discouraged-namespace` should trigger warnings regardless of whether those namespaces appear in any groups or not. 

Motivating example: 

Suppose I have two groups, `metabase.db.*` (`db`) and `metabase.api.*` (`api`). I want to disallow anything in the `db` group from accessing the `metabase.api.internal` namespace. That doesn't work however.

I think what's going on is that once a namespace is grouped into a group in `:ns-groups`, it basically ceases to exist from that point out as far as Kondo namespace-related config is concerned. I'm not sure whether this was intentional behavior, but if it was, I think it's counterintuitive. 

I could definitely see use cases for having a namespace match multiple patterns and get some sort of combined config from several groups. Perhaps namespace A and B would get config 1, and B and C get config 2 -- if namespaces matching multiple groups worked, that's pretty easy. Without multiple groups, you would have to have three separate configs, and the B config would just be a copy of configs 1 + 2 -- that gets hard to do when you start having a lot of groups and a big config file.

I couldn't really find anything in the dox about whether this behavior is intentional or not. Either way if it turns out to be technically infeasible or something, I still think plain namespaces should still work in `:discouraged-namespace` regardless of which groups that namespace might appear in

**Comment by camsaul** [comment]
I think #1945 is part of this issue

**Comment by camsaul** [comment]
This also seems to affect `:discouraged-var` -- if the namespace of the `:discouraged-var` matches **any** namespaces in the `:ns-groups`, the rule stops working.
